src/common/AndroidUtils.ts

```diff
@@ -504,29 +529,249 @@ export class AndroidUtils {
                         resolvedEmulator,
                         requestedPortNumber
                     );
-                    return Promise.resolve(portNumber);
+
+                    const isWritable = await AndroidUtils.isEmulatorSystemWritable(
+                        portNumber
+                    );
+
+                    if (isWritable === writable) {
+                        // already launched and modes match so just return its port
+                        return Promise.resolve(portNumber);
```

Comment:
The logic here ensure that if an emulator is already running then it must be running in the same "mode" as what is passed to this function. So if it is running with `writable-system` but this function is called with `writable=false` then that's a mismatch and we would shut down & relaunch.

Should we leave it this way or should we only care for the scenario where `writeable=true` but `isWritable=false`? In other words if an emulator is running with `writable-system` but this function is called with `writable=false` should we ignore and continue or should I do what I do now and shut it down + relaunch?